blog_post_collaboration/kedro_ibis.md

```diff
@@ -0,0 +1,263 @@
+
+```bash
+kedro run --pipeline seed
+```
+
+Finally, run the actual pipeline:
+
+```bash
+kedro run
+```
+
+Voilà! Feel free to confirm that the expected tables and views got created:
+
+```pycon
+>>> import duckdb
+>>>
+>>> con = duckdb.connect("jaffle_shop.duckdb")
+>>> con.sql("SHOW TABLES")
+┌───────────────┐
+│     name      │
+│    varchar    │
+├───────────────┤
+│ customers     │
+│ orders        │
+│ raw_customers │
+│ raw_orders    │
+│ raw_payments  │
+│ stg_customers │
+│ stg_orders    │
+│ stg_payments  │
+└───────────────┘
+
+>>> con.sql("SELECT * FROM customers WHERE customer_id = 42")
+┌─────────────┬────────────┬───────────┬─────────────┬───────────────────┬──────────────────┬─────────────────────────┐
+│ customer_id │ first_name │ last_name │ first_order │ most_recent_order │ number_of_orders │ customer_lifetime_value │
+│    int64    │  varchar   │  varchar  │    date     │       date        │      int64       │         double          │
+├─────────────┼────────────┼───────────┼─────────────┼───────────────────┼──────────────────┼─────────────────────────┤
+│          42 │ Diana      │ S.        │ 2018-02-04  │ 2018-03-12        │                2 │                    27.0 │
+└─────────────┴────────────┴───────────┴─────────────┴───────────────────┴──────────────────┴─────────────────────────┘
+
+```
+
+As always, we can view the pipeline using Kedro-Viz:
+
+<img width="1470" alt="image" src="https://github.com/kedro-org/kedro-devrel/assets/14007150/9fedaf00-ecf6-418f-9c1e-a98844994ba2">
+
+## But wait... there's more!
+
+Making pipeline productionisation easier alone may justify adopting Ibis in your Kedro workflows, but other situations from my past experience also come to mind.
+
+Have you ever found yourself developing data pipelines against an existing data warehouse, even though you know that the data infrastructure team is currently migrating to a different database solution? You expect to rewrite some amount of your code—unless you use Ibis?
+
+Leveraging Ibis can also help you build truly reusable pipelines. I previously led development of a Kedro-based code asset that provided a suite of reusable pipelines for customer analytics. We decided to use PySpark for data engineering, but many of our users didn't need to use PySpark. One major retailer stored all of their data in Oracle, and they ended up with a suboptimal workflow wherein they extracted data into Spark and did all the work there. Other teams looked at our pipelines for inspiration, but ended up rewriting them for their infrastructure—not our intention in delivering prebuilt pipelines. Last but not least, we spent so much time and effort setting up Spark on locked-down Windows machines, so data scientists could play with the pipelines we provided; being able to run the same logic using DuckDB or pandas locally would have been a godsend!
+
+## What's next?
```

Comment:
Love the post, especially about what values/benefits Kedro on Ibis can offer. It may be valuable to include a section for tradeoffs or current limitations. That way, your audience knows you are not biased.

Reply:
Agree completely - 

Some limitations off the top of my head:

- I think we can highlight that the talent bar for a SQL engineer is much lower. The SQL equivalents of venvs, dependency management and pytest are all trivial in comparison. 
- The dbt / jinja tech debt interest payments may be reasonable depending on your sophistication, pace of development and standards enforcement.
- If you are in an environment where the tech stack has been decided and locked down, flexibility isn't an advantage and using native tooling may be a better bet.

Reply:
@deepyaman I shall leave you to address & resolve the feedback so far. Please ping me here when you're ready for me to take a look at your next draft.

Reply:
Hmm, good point.

> Some limitations off the top of my head:
>
> * I think we can highlight that the talent bar for a SQL engineer is much lower. The SQL equivalents of venvs, dependency management and pytest are all trivial in comparison.
> * The dbt / jinja tech debt interest payments may be reasonable depending on your sophistication, pace of development and standards enforcement.
> * If you are in an environment where the tech stack has been decided and locked down, flexibility isn't an advantage and using native tooling may be a better bet.

I feel these are less limitations of using Ibis or integrating Kedro and Ibis, and more the Python vs. SQL debate (which I don't want to enter). The goal isn't to convert dbt users to Kedro, but rather to make Kedro users write better pipelines. 

Some ideas that come to mind though, in this vein:

* In general, I think DDL support in Ibis is relatively newer/less fleshed out in Ibis (@cpcloud is this still true? Is this something the sqlglot migration will help significantly in?). Right now, the integration also supports much fewer materializations than dbt does (only `table` and `view`, no `materialized view` or `incremental`).
* On the topic of materializations, dbt also supports `ephemeral`. The Kedro equivalent is `MemoryDataset`, but one thing I didn't do is somehow set default copy mode to assign (like it is for Spark), which would be necessary for it to work correctly. Maybe I can create an issue for this, and link to it?
* dbt has the ability to detect and deploy only modified models; Kedro doesn't have anything automatic like that.

Reply:
Fair ponts - 

> dbt has the ability to detect and deploy only modified models; Kedro doesn't have anything automatic like that.

This is reasonably easy to implement if we make the reasonable assumption that the user is working in a git repo